requests/adapters.py

```diff
@@ -418,11 +421,14 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                        "timeout tuple, or a single float to set "
                        "both timeouts to the same value".format(timeout))
                 raise ValueError(err)
-        else:
+        elif timeout is not None:
             timeout = TimeoutSauce(connect=timeout, read=timeout)
```

Comment:
Unfortunately, this change is breaking. You'll either need to remove the default timeout change, or move this PR to the 3.0 branch.

You're also welcome to split this PR into two parts, the first bit making just the non-breaking changes against the master branch and the second bit making the breaking ones against 3.0.